OCPBUGS-58103: Disable LightSpeed dialog on non amd64 architecture

Leo6Leo opened the pull request:
## Problems
The Lightspeed icon pops up in the OpenShift UI in 4.19.1. Can we either get support for OpenShift Lightspeed on Power or have this dialog removed? It sends weird messages.

## Changes
- Add the new logic to set the compatibility properties based on the cluster environment
- Unit tests updated to expect the correct compability values

## Impacts
- On non homogenious amd64 cluster, the lightspeed icon won't show up in the dashboard.

Leo6Leo commented on pkg/console/subresource/consoleserver/config_builder.go:
```diff
@@ -601,3 +602,61 @@ func (b *ConsoleServerCLIConfigBuilder) proxy() Proxy {
+	// Only enable Lightspeed if ALL nodes are AMD64 (homogeneous cluster)
+	// nodeArchitectures contains deduplicated architectures from all nodes
+	var isHomogeneousAMD64 bool
+	if len(b.nodeArchitectures) == 1 {
```
I think the `nodeArchitectures` variable already contains the deduplicated architecture types, so checking its length is all we need.

If `nodeArchitectures.length > 1`, it immediately tells us that the cluster is heterogeneous because it includes more than one distinct architecture. Therefore imo i think the loop here should work.

Reference: 
1. [Where the values of nodeArchitectures being assigned](https://github.com/openshift/console-operator/blob/main/pkg/console/operator/sync_v400.go#L354)

2. [The definition of the getNodeComputeEnvironments function that is being called when assigning the value to nodeArchitectures](https://github.com/openshift/console-operator/blob/main/pkg/console/operator/sync_v400.go#L733C1-L752C2)

Leo6Leo commented on pkg/console/subresource/consoleserver/config_builder.go:
```diff
@@ -601,3 +602,43 @@ func (b *ConsoleServerCLIConfigBuilder) proxy() Proxy {
+		klog.V(4).Infof("enabling LightspeedButton capability for homogeneous AMD64 cluster")
+	}
+
+	if len(capabilities) > 0 {
```
I kept the condition check here because, in the test suite, we don't set nodeArchitecture, nor do we provide a default for the capability. As you mentioned, the default capabilities are defined in the YAML file [here](https://github.com/openshift/console-operator/blob/main/manifests/01-operator-config.yaml#L18). Since the test environment doesn't have access to this YAML, some tests may fail due to the lack of these default values.

Please let me know if you have any suggestions for a better approach to resolve this issue. Thanks!

jhadvig replied:
I think its ok.
But just to be sure I would rather travers the array and find the appropriate lightspeed item, set it and print the value for the current item so we can drop the line below with the logging

jhadvig reviewed (commented):
Thanks Leo 👍 
Adding couple of comments.

Leo6Leo commented on pkg/console/operator/sync_v400_test.go:
```diff
@@ -212,3 +213,77 @@ func TestGetNodeComputeEnvironments(t *testing.T) {
+		expectedOperatingSystems []string
+	}{
+		{
+			name: "1000 AMD64 nodes - should return single architecture",
```
> Also not really sure why we need additional test, cause there is TestGetNodeComputeEnvironments() above

I've now merged the two functions to avoid redundancy! I originally created this separate test function here to improve readability and make it easier to understand in the context of recent changes.

Leo6Leo commented:
/cc @jhadvig I've addressed the review comment and also refactored the code slightly to make future updates easier. Now, if we add support for additional architectures, we can simply extend the string without needing to significantly modify the existing codebase. PTAL!

Leo6Leo commented on pkg/console/subresource/consoleserver/config_builder.go:
```diff
@@ -601,3 +606,42 @@ func (b *ConsoleServerCLIConfigBuilder) proxy() Proxy {
+	}
+
+	// Check if the architecture is in the supported list
+	clusterArch := b.nodeArchitectures[0]
```
ohh actually that's true, i didn't catch that, thanks!! The fix has been applied.

jhadvig commented:
/lgtm
